Arm/Insts/DPSFP/Advanced_simd_modified_immediate.lean

```diff
@@ -91,46 +94,50 @@ private theorem mul_div_norm_form_lemma  (n m : Nat) (_h1 : 0 < m) (h2 : n ∣ m
 -- Assumes CheckFPAdvSIMDEnabled64();
 def exec_advanced_simd_modified_immediate
   (inst : Advanced_simd_modified_immediate_cls) (s : ArmState) : ArmState :=
-  if inst.cmode == 0b1111#4 && inst.op == 0b0#1 && inst.o2 == 0b1#1 then
-    write_err (StateError.Unimplemented s!"Unsupported {inst} encountered!") s
-  else if inst.cmode == 0b1111#4 && inst.op == 0b1#1 && inst.Q == 0b0#1 || inst.o2 == 0b1#1 then
+  -- All UNALLOCATED cases when inst.o2 = 1
+  if inst.o2 == 0b1#1 ∧ inst.cmode ++ inst.op != 0b11110#5 then
```

Comment:
Could this check also be done in `decode_immediate_op`, since this function has access to `inst` anyway?

Reply:
Sure. Let me try.